code/modules/mob/living/carbon/carbon.dm

```diff
@@ -127,7 +127,10 @@
 
 
 /mob/living/carbon/proc/do_vomit()
-	Stun(10 SECONDS)
+
+	adjust_stagger(3) //A bit of soft CC instead of an utterly ridiculous 10 second stun; baycode remnants be gone; we'll see how this plays with Transvitox
+	add_slowdown(3)
+
```

Comment:
How long is this
Remove the comment

Reply:
5 seconds on average as it processes on life tick, and I assume an average of 1 second (between 0 and 2) for the first tick (2 * 2 + 1)